data-prepper-plugins/aws-lambda/src/integrationTest/java/org/opensearch/dataprepper/plugins/lambda/processor/LambdaProcessorServiceIT.java

```diff
@@ -0,0 +1,165 @@
+    @Mock
+    private PluginMetrics pluginMetrics;
+    @Mock
+    private PluginFactory pluginFactory;
+    @Mock
+    private PluginSetting pluginSetting;
+    @Mock
+    private Counter numberOfRecordsSuccessCounter;
+    @Mock
+    private Counter numberOfRecordsFailedCounter;
+    @Mock
+    private ExpressionEvaluator expressionEvaluator;
+
+    private final ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory().enable(YAMLGenerator.Feature.USE_PLATFORM_LINE_BREAKS));
+
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        MockitoAnnotations.openMocks(this);
+        lambdaRegion = System.getProperty("tests.lambda.processor.region");
+        functionName = System.getProperty("tests.lambda.processor.functionName");
+        role = System.getProperty("tests.lambda.processor.sts_role_arn");
+
+        final Region region = Region.of(lambdaRegion);
+
+        lambdaClient = LambdaClient.builder()
+                .region(Region.of(lambdaRegion))
+                .build();
+
+        bufferFactory = new InMemoryBufferFactory();
+
+        when(pluginMetrics.counter(LambdaProcessor.NUMBER_OF_RECORDS_FLUSHED_TO_LAMBDA_SUCCESS)).
+                thenReturn(numberOfRecordsSuccessCounter);
+        when(pluginMetrics.counter(LambdaProcessor.NUMBER_OF_RECORDS_FLUSHED_TO_LAMBDA_FAILED)).
+                thenReturn(numberOfRecordsFailedCounter);
+    }
+
+
+    private static Record<Event> createRecord() {
+        final JacksonEvent event = JacksonLog.builder().withData("[{\"name\":\"test\"}]").build();
+        return new Record<>(event);
+    }
+
+    public LambdaProcessor createObjectUnderTest(final String config) throws JsonProcessingException {
+
+        final LambdaProcessorConfig lambdaProcessorConfig = objectMapper.readValue(config, LambdaProcessorConfig.class);
+        return new LambdaProcessor(pluginMetrics,lambdaProcessorConfig,awsCredentialsSupplier,expressionEvaluator);
+    }
+
+    public LambdaProcessor createObjectUnderTest(LambdaProcessorConfig lambdaSinkConfig) throws JsonProcessingException {
+        return new LambdaProcessor(pluginMetrics,lambdaSinkConfig,awsCredentialsSupplier,expressionEvaluator);
+    }
+
+
+    private static Collection<Record<Event>> generateRecords(int numberOfRecords) {
+        List<Record<Event>> recordList = new ArrayList<>();
+
+        for (int rows = 1; rows <= numberOfRecords; rows++) {
+            HashMap<String, String> eventData = new HashMap<>();
+            eventData.put("name", "Person" + rows);
+            eventData.put("age", Integer.toString(rows));
+
+            Record<Event> eventRecord = new Record<>(JacksonEvent.builder().withData(eventData).withEventType("event").build());
+            recordList.add(eventRecord);
+        }
+        return recordList;
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {1,3})
+    void verify_records_to_lambda_success(final int recordCount) throws Exception {
```

Comment:
consider adding test for `InvocationType` Event

Reply:
invocation type event will be disabled for now, will be releasing event type with asynchronous support that requires additional infra changes. I have disabled it in the verification for now, ll fix the readme.